app/helpers/items_helper.rb

```diff
@@ -1,8 +1,10 @@
 # frozen_string_literal: true
 
 module ItemsHelper
-  def stacks_url_full_size(druid, file_name)
-    "#{Settings.stacks_file_url}/#{druid}/#{ERB::Util.url_encode(file_name)}"
+  def stacks_url_full_size(druid, filename)
+    # Allow literal slashes in the filename (do not encode them)
+    encoded_filename = filename.split('/').map { |filename_part| ERB::Util.url_encode(filename_part) }.join('/')
```

Comment:
The same approach is already used in [sul-embed](https://github.com/sul-dlss/sul-embed/pull/1376/files#diff-bbb9f89625df7768bb24af8d2af0b559798c229837899cbaf3c67d7091c506feR39)